addons/mrp/tests/test_order.py

```diff
@@ -3014,6 +3014,82 @@ def test_replan_workorders02(self):
         self.assertFalse(wo_02.show_json_popover)
         self.assertEqual(wo_01.date_planned_finished, wo_02.date_planned_start)
 
+    @freeze_time('2022-10-21 12:00')
+    def test_replan_mo_without_bom(self):
+        """
+            Create 2 MOs without BoM
+                just set the product and a component
+
+            For first MO :
+                Add 2 WO (with different WC)
+                Don't schedule first WO
+                Schedule second WO
+                Confirm => MO is Confirmed and Planned
+                Schedule first WO before second WO
+                Confirm => MO should Replan without Error
+
+            For second MO :
+                Add 1 Scheduled WO
+                Confirm => MO is Confirmed and Planned
+                Add a second WO scheduled before the other one (with different WC)
+                Confirm => MO should Replan without Error
+        """
+
+        # Required for `workorder_ids` to be visible in the view
+        self.env.user.groups_id += self.env.ref('mrp.group_mrp_routings')
+
+        mos = self.env['mrp.production']
+        for _ in range(2):
+            mo_form = Form(self.env['mrp.production'])
+            mo_form.product_id = self.product_8
+            with mo_form.move_raw_ids.new() as component:
+                component.product_id = self.product_6
+            mos += mo_form.save()
+        mo_01, mo_02 = mos
+
+        #First MO
+        with Form(mo_01) as mo_01_form:
+            with mo_01_form.workorder_ids.new() as workorder:
+                workorder.name = "OP1"
+                workorder.workcenter_id = self.workcenter_2
+            with mo_01_form.workorder_ids.new() as workorder:
+                workorder.name = "OP2"
+                workorder.workcenter_id = self.workcenter_3
+                workorder.date_planned_start = datetime(2022, 10, 23, 12)
+            mo_01 = mo_01_form.save()
+        mo_01.action_confirm()
+
+        with Form(mo_01) as mo_01_form:
+            with mo_01_form.workorder_ids.edit(0) as workorder:
+                workorder.date_planned_start = datetime(2022, 10, 22, 12)
+            mo_01 = mo_01_form.save()
+
+        try:
+            mo_01.action_confirm()
+        except TypeError:
+            self.fail("mo_01.action_confirm raised TypeError unexpectedly.")
```

Comment:
I've specifically write these lines to make the assertion explicit and related to the error that was first raised before this fix.
I may replace these by a comment though

Reply:
The purpose of the tests is to write the expected correct flow so that way if a test fails, we know something is wrong in whatever code change we're trying to do. It's confusing for others if a test takes into account a failure that shouldn't happen. It is indeed standard to include the original error in the fix commit message (also the steps for how to reproduce the error before the fix was applied.)